src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -149,6 +174,58 @@ internal static byte[][] SegmentedEncodeUnsignedInteger(byte[] bigEndianBytes, i
         }
 
         /// <summary>
+        /// Encode the segments { tag, length, value } of a bit string where all bits are significant.
+        /// </summary>
+        /// <param name="data">The data to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeBitString(byte[] data)
+        {
+            return SegmentedEncodeBitString(0, data);
+        }
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of a bit string where the least significant
+        /// <paramref name="unusedBits"/> of the last byte are padding.
+        /// </summary>
+        /// <param name="unusedBits">The number of padding bits (0-7) in the last byte</param>
+        /// <param name="data">The data to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeBitString(int unusedBits, byte[] data)
+        {
+            Debug.Assert(data != null);
+            Debug.Assert(unusedBits >= 0);
+            Debug.Assert(unusedBits <= 7);
+            Debug.Assert(unusedBits == 0 || data.Length > 0);
+
+            byte[] encodedData = new byte[data.Length + 1];
+
+            // Copy data to encodedData, but leave a one byte gap for unusedBits.
+            Buffer.BlockCopy(data, 0, encodedData, 1, data.Length);
+            encodedData[0] = (byte)unusedBits;
+
+            // We need to make a mask of the bits to keep around for the last
+            // byte, ensuring we clear out any bits that were set, but reported
+            // as padding by unusedBits
+            // 
+            // For example:
+            // unusedBits 0 => mask 0b10000000
+            // unusedBits 1 => mask 0b11000000
+            // unusedBits 7 => mask 0b11111111
+            // unusedBits 8 => mask 0b10000000
+            byte lastByteSemanticMask = unchecked((byte)(-1 << unusedBits));
```

Comment:
Oops, copied it from the NamedBitList where the input value had the opposite semantics.  Will fix.